lib/spack/spack/cmd/create_tarball.py

```diff
@@ -43,11 +43,24 @@ def create_tarball(parser, args):
     if not args.packages:
         tty.die("tarball creation requires at least one package argument")
 
-    specs = spack.cmd.parse_specs(args.packages, concretize=True)
+    specs = spack.cmd.parse_specs(args.packages)
     for spec in specs:
-        package = spack.repo.get(spec)
+        # ----------- Make sure the spec only resolves to ONE thing
+        q = spack.installed_db.query(spec)
```

Comment:
This package = spack.repo.get(spec) gives you the abstract spec.
This q = spack.installed_db.query(spec) give you the installed spec.
Somehow this produces two different hashes.